MLIR.cmake

```diff
@@ -1,10 +1,17 @@
 # SPDX-License-Identifier: Apache-2.0
 
+# Must unset LLVM_DIR in cache. Otherwise, when MLIR_DIR changes LLVM_DIR
+# won't change accordingly.
+unset(LLVM_DIR CACHE)
 if (DEFINED ENV{MLIR_DIR})
-  set(MLIR_DIR $ENV{MLIR_DIR} CACHE PATH "Path to directory containing MLIRConfig.cmake")
+  set(MLIR_DIR $ENV{MLIR_DIR} CACHE PATH "Path to directory containing MLIRConfig.cmake" FORCE)
```

Comment:
I would prefer to drop the environment var option entirely rather than make it take precedence always. I understand this is unlikely to happen, but I think it's a good idea to consider not using environment variables as much as possible since generally cmake expects to work with configuration options (that are then cached, so you never have to pass them more than once).

Reply:
Done.